src/pam_google_authenticator.c

```diff
@@ -574,6 +578,52 @@ full_write(int fd, const char* buf, size_t len) {
   }
 }
 
+static int set_selinux_context(int fd) {
+  char *old_context = NULL;
+  int err = 0;
+  
+  // skip if SELinux is not enabled
+  if (!is_selinux_enabled()) {
+    err = 1;
+    goto cleanup;
```

Comment:
Are these gotos safe? They jump past `ctx` initialization and calls `context_free()` on `ctx`. Maybe `ctx` should be defined up with `old_context` to something safe to free?